### DISCUSSION TITLE
InputControllerWin32 Mouse wheel

### DISCUSSION THREAD

**Discussion by Floady** [DiligentGraphics/DiligentSamples#111] [General]
When the mouse wheel value is updated, you probably also want to set the flag

```
 case WM_MOUSEWHEEL:
            // Update member var state
            m_MouseState.WheelDelta = (float)((short)HIWORD(wParam)) / (float)WHEEL_DELTA;
            m_MouseState.ButtonFlags |= MouseState::BUTTON_FLAG_WHEEL;
            MsgHandled              = true;
            break;
```

**Comment by TheMostDiligent** [comment]
It is probably not necessary. The flags are used to detect when L/B/R mouse button is pressed and released. For the wheel, the `WheelDelta` already conveys the information.
Your change will set the `BUTTON_FLAG_WHEEL` flag, but it will never be cleared and will always remain set.
